Update README.md (#470)
val-lang/val: README.md

- `--emit raw-ir`: Lower the typed AST into Val IR and output the result in a file.
- `--emit ir`: Run mandatory IR passes and output the result in a file.
- `--emit cpp`: Produce a C++ source file.

> - `--emit binary` (default): Produce an executable (At this stage Val need to compile to C++ first).

Reviewer: ```suggestion
- `--emit binary` (default): Produce an executable (currently by compiling transpiled C++ files)
```
